### ISSUE TITLE
FTP Server not reachable until app restart

### ISSUE THREAD

**Issue by praecipitator** [wolpi/prim-ftpd#365]
It is somewhat difficult to phrase the problem properly, so I might have missed an already open issue about that. Apologies if I did.

Now, the issue itself:

When I want to use PrimtiveFTPd, it usually doesn't initially work. I open the app, tap the "play" button, but my FTP clients can't actually connect to it. Filezilla outputs a "FATAL ERROR: No route to host". However, if I then either force-quit the app, or open that "overview" of current apps and swipe it away, and then start it again, everything works fine.

The phone is an OnePlus 5T, running a rooted LineageOS 21-20240611-NIGHTLY-dumpling (but I update it rather often), which is based on Android 14. PrimitiveFTPd I got off F-Droid, version "7.2 (code: 65)", but it was the same on the previous version as well. The FTP client I've been using is FileZilla on Linux and Windows, over my local Wi-Fi. 
I set the app to "good old file system", and I use SFTP with a username/password.

**Comment by wolpi** [comment]
Sounds weird. I never had such an issue myself. Not sure if there is something we can do about that.

**Comment by lmagyar** [comment]
FYI, I have similar/related? issues, though never reported.

In my case after I change any config, before I restart the sftp service (stop + start button), I have to change to another screen in the app (ie. swipe left or right), then stop + start. Without the swipe, it doesn't notice the config change when I press stop + start.

The only exception was (as I remember) the Server start dir, if i change that, I have to do the same as OP, ie. close the app completely and restart again.

I'm on Anfroid 9.

**Comment by wolpi** [comment]
Hmm, that sounds as it could be related to our code. Let's see if we can figure out something, and when ...

**Comment by lmagyar** [comment]
pinging @hamidsafdari: I see you are fixing some UI/Android related issues now, maybe you have a clue what's the root problem here

TLDR: it seems, that the settings are stored only after we switch to another screen/activity (I don't know the proper Andorid terminology), so after we modified a setting and before we restart the server, we have to change to another screen/activity to get these changes noticed by the app, and maybe not all Android versions show this issue.

**Comment by hamidsafdari** [comment]
It looks like the context of the PFTPD tab fragment does not get updated when the preferences are changed. When the start icon in the toolbar is tapped, it's this fragment that's passed to the start function. The context of the fragment is then used to the get the preferences which are outdated.

The fix was to send the activity context to the start function. Tested this on APIs 26 and 34.

**Comment by lmagyar** [comment]
Thats absolutely fantastic! :) I've merged it into my fork, and it works (API 28, Android 9).

It made me mad, to always swipe, forget it, restart. But I know nearly nothing about Android internals.

And my previous comment about start dir also fixed, I think my extra issue was, that WinSCP remembered the previous home dir, so I had to close the session in WinSCP also. And now I see the new start dir, everything is OK on my side. :)
